src/Microsoft.Data.SqlClient/src/Microsoft/Data/SqlClient/ConnectionPool/DbConnectionPool.cs

```diff
@@ -0,0 +1,120 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Concurrent;
+using System.Data.Common;
+using System.Threading.Tasks;
+using System.Transactions;
+using Microsoft.Data.Common;
+using Microsoft.Data.ProviderBase;
+
+namespace Microsoft.Data.SqlClient.ConnectionPool
+{
+    internal abstract class DbConnectionPool
+    {
+        private static int _objectTypeCount; // EventSource counter
+        internal readonly int _objectID = System.Threading.Interlocked.Increment(ref _objectTypeCount);
+
+        internal int ObjectID
+        {
+            get
+            {
+                return _objectID;
+            }
+        }
+
+        private DbConnectionPoolState _state;
+        internal DbConnectionPoolState State
+        {
+            get => _state;
+            set => _state = value;
+        }
+
+        #region Abstract Properties
+        internal abstract int Count
+        {
+            get;
```

Comment:
nit: These should be on the same line as the definition (ie `internal abstract int Count { get; }`)